PresentScreenings/Utilities/WebUtility.cs

```diff
@@ -177,25 +193,81 @@ public static FilmInfo TryParseUrlSummary(HttpWebRequest request, string url, Me
                     }
                     success = true;
                 }
+
+                // Assert if there were no matches for this segment.
                 if (!success)
                 {
                     allParsesFailed = true;
                 }
             }
+
+            // Throw an exception if at least one of the expected segments didn't match.
             if (allParsesFailed)
             {
-                builder.AppendLine($"URL {url} could not be parsed.");
-                builder.AppendLine("===");
-                builder.Append(text);
-                throw new UnparseblePageException(builder.ToString());
+                throw new UnparseblePageException(text);
             }
+
+            // Create a Film Info instance with the information found.
             var info = new FilmInfo(filmId, Film.FilmInfoStatus.Complete, filmDescription, article);
             foreach (string title in ScreenedFileDescriptionByTitle.Keys)
             {
                 info.AddScreenedFilm(title, ScreenedFileDescriptionByTitle[title]);
             }
+
             return info;
         }
+
+        public static async Task<bool> VisitUrl(Film film, CancellationToken cancellationToken)
+        {
+            MediumCatagory catagory = film.Catagory;
+            string url = film.Url;
+            bool canceled = false;
+            bool webErrorOccurred = false;
+            WebClient httpClient = new WebClient();
+            string contents = string.Empty;
+
+            // Read the website of the given film.
+            try
+            {
+                // Create an task to asynchroneously read the website.
+                Task<string> contentsTask = httpClient.DownloadStringTaskAsync(url);
+
+                // Return control to the calling code until the asynchronous
+                // task finishes on its own thread.
+                contents = await contentsTask;
+
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+            catch (OperationCanceledException)
+            {
+                canceled = true;
+            }
```

Comment:
Good indication to use "canceled" instead of "cancelled".
Though in the same context "cancellation" is used in favour of "cancelation".